backend/pkg/web/middleware/require_auth.go

```diff
@@ -31,17 +34,40 @@ func RequireAuth() gin.HandlerFunc {
 			c.Abort()
 			return
 		}
-		claim, err := auth.JwtValidateFastenToken(appConfig.GetString("jwt.issuer.key"), tokenString)
+		
+		// Parse token using existing JWT function
+		claims, err := auth.JwtValidateFastenToken(appConfig.GetString("jwt.issuer.key"), tokenString)
 		if err != nil {
 			c.JSON(http.StatusUnauthorized, gin.H{"success": false, "error": err.Error()})
 			c.Abort()
 			return
 		}
-
-		//todo, is this shared between all sessions??
+		
+		// Check if it's an access token
+		if claims.TokenType == "access" {
+			databaseRepo := c.MustGet(pkg.ContextKeyTypeDatabase).(database.DatabaseRepository)
+			token, err := databaseRepo.GetAccessToken(c, claims.TokenID)
+			if err != nil || token == nil {
+				c.JSON(http.StatusUnauthorized, gin.H{"success": false, "error": "Invalid access token"})
+				c.Abort()
+				return
+			}
+			hashedToken := models.HashToken(tokenString)
+			if hashedToken != token.TokenHash {
+				c.JSON(http.StatusUnauthorized, gin.H{"success": false, "error": "Invalid access token"})
+				c.Abort()
+				return
+			}
+			if token.IsExpired() {
+				c.JSON(http.StatusUnauthorized, gin.H{"success": false, "error": "Access token is expired"})
+				c.Abort()
+				return
+			}
+		}
```

Comment:
this validation logic feels like it should live in `JwtValidateFastenToken` or similar